nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -93,93 +70,47 @@ private static Random getRandomnessSource(String source) {
         }
 
         List<T> newOrder = Arrays.asList(originalOrder);
-        newOrder = ControlNondeterminism.internalShuffle(newOrder, ControlNondeterminism.getSource());
+
+        newOrder = nondex.permutation(newOrder);
+
         newOrder.toArray(originalOrder);
 
         // return in place
         return originalOrder;
```

Comment:
With List it returns new List, while with array it returns the original modified array. Is there a reason they should be different in this regard?